spec/lib/record_loader/barcode_printer_type_loader_spec.rb

```diff
@@ -5,10 +5,12 @@
 
 # This file was initially generated via `rails g record_loader`
 RSpec.describe RecordLoader::BarcodePrinterTypeLoader, type: :model, loader: true do
-  subject(:record_loader) do
+  def a_new_record_loader
     described_class.new(directory: test_directory, files: selected_files)
   end
 
```

Comment:
When attempting to reproduce the issue in tests I couldn't, as it was using the same instance of record-loader.
I've ported the updated behaviour to all other record loaders, just in case we were missing other similar bugs.
I'll also update the template in record loader itself.